backend/controller/observability/pubsub.go

```diff
@@ -0,0 +1,76 @@
+package observability
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/metric"
+	"go.opentelemetry.io/otel/metric/noop"
+
+	"github.com/TBD54566975/ftl/backend/schema"
+	"github.com/TBD54566975/ftl/internal/observability"
+)
+
+const (
+	pubsubMeterName                = "ftl.pubsub"
+	pubsubTopicNameAttribute       = "ftl.pubsub.topic.name"
+	pubsubSubscriptionRefAttribute = "ftl.pubsub.subscription.ref"
+	pubsubSubscriberRefAttribute   = "ftl.pubsub.subscriber.sink.ref"
+)
+
+type PubSubMetrics struct {
+	meter            metric.Meter
+	published        metric.Int64Counter
+	subscriberCalled metric.Int64Counter
+}
+
+func initPubSubMetrics() (*PubSubMetrics, error) {
+	result := &PubSubMetrics{}
+	var errs error
+	var err error
+
+	result.meter = otel.Meter(pubsubMeterName)
+
+	counterName := fmt.Sprintf("%s.published", pubsubMeterName)
+	if result.published, err = result.meter.Int64Counter(
+		counterName,
+		metric.WithUnit("1"),
+		metric.WithDescription("the number of times that an event is published to a topic")); err != nil {
+		errs = handleInitCounterError(errs, err, counterName)
```

Comment:
What's the purpose of chaining errors vs. returning the first error? I suspect if anything fails in metric construction, it will be some underlying issue that will be identical across all of them, so it's unlikely you're going to get additional useful information out of the system.

Reply:
That's definitely the most likely case, but since it's not an absolute guarantee, we thought this would be the safer route to start out with.